tests/test_transformation.py

```diff
@@ -260,6 +260,114 @@ def test_dependency_transformation_module_imports(frontend):
     assert driver['driver_mod'].spec.body[0].module == 'kernel_test_mod'
     assert 'kernel_test' in [str(s) for s in driver['driver_mod'].spec.body[0].symbols]
 
+@pytest.mark.parametrize('frontend', available_frontends())
+def test_dependency_transformation_globalvar_imports(frontend):
+    """
+    Test that global variable imports are not renamed as a
+    call statement would be.
+    """
+
+    kernel = Sourcefile.from_source(source="""
+MODULE kernel_mod
+    INTEGER :: some_const
+CONTAINS
+    SUBROUTINE kernel(a, b, c)
+    INTEGER, INTENT(INOUT) :: a, b, c
+
+    a = 1
+    b = 2
+    c = 3
+  END SUBROUTINE kernel
+END MODULE kernel_mod
+""", frontend=frontend)
+
+    driver = Sourcefile.from_source(source="""
+SUBROUTINE driver(a, b, c)
+    USE kernel_mod, only: kernel
+    USE kernel_mod, only: some_const
+    INTEGER, INTENT(INOUT) :: a, b, c
+
+    CALL kernel(a, b ,c)
+END SUBROUTINE driver
+""", frontend=frontend)
+
+    transformation = DependencyTransformation(suffix='_test', module_suffix='_mod')
+    kernel.apply(transformation, role='kernel')
+    driver.apply(transformation, role='driver', targets=('kernel', 'some_const'))
```

Comment:
Just to chime in here (I still haven't looked through all the changes, sorry!): In the Scheduler rewrite we can (and want) to consider also other dependency types than control flow, which therefore become "live edges". And considering the scope of the transformation (traverse the dependency tree to global variable leaf nodes), I think it is actually in line with the concept of "targets" - filtering these symbols from the `targets` list when not required by a tree traversal should become a functionality of the SFilter class.

Reply:
Yeah I add `GlobalVarImportItem`s as targets so that within the transformation I can filter out the imports I am interested in based on the `disable\ignore\block` lists in the scheduler config.